src/karmabot/commands/note.py

```diff
@@ -0,0 +1,113 @@
+import datetime
+import re
+from typing import Callable, Dict, Union
+
+from karmabot.db import db_session
+from karmabot.db.karma_note import KarmaNote
+from karmabot.db.karma_user import KarmaUser
+
+NOTE_CMD_PATTERN = re.compile(r"note\s(\w+)\s?(.*)")
+NOTE_COMMANDS: Dict[str, Callable] = {
+    "add": lambda text_arg, user: _add_note(text_arg, user),
+    "del": lambda text_arg, user: _del_note(text_arg, user),
+    "list": lambda text_arg, user: _list_notes(text_arg, user),
+    "default": lambda text_arg, user: f"Sorry {user.username}, your note command was not recognized. You can use {', '.join(NOTE_COMMANDS.keys())}.",

```

Comment:
long, complex line, probably better to move outside of this mapping, making it its own constant

Reply:
so yes the problem is I get an "not defined" error...I guess this is because dictionarys work differently then using a function inside another function...not quite sure I do some research

Reply:
OK it is really a look ahead problem which I never encountered before in Python...so without the lambda you HAVE to put the dict at the end of the file so that Python knows about the referred functions, otherwise you get an not found error. However, this is not the case with lambdas because they are not evaluated....so that is due to compiler check versus runtime check